cgi-bin/DW/Controller/OpenID.pm

```diff
@@ -22,10 +23,41 @@ use DW::Routing;
 use DW::Controller;
 use DW::Template;
 
+# for responding to OpenID authentication requests
+DW::Routing->register_string( '/openid/server', \&openid_server_handler,
+                                                app => 1, no_cache => 1 );
+
+# for claiming imported comments
 DW::Routing->register_string( '/openid/claim', \&openid_claim_handler, app => 1 );
 DW::Routing->register_string( '/openid/claimed', \&openid_claimed_handler, app => 1 );
 DW::Routing->register_string( '/openid/claim_confirm', \&openid_claim_confirm_handler, app => 1 );
 
+sub openid_server_handler {
+    return "OpenID consumer support is disabled"
+        unless LJ::OpenID::server_enabled();
+
+    my $r = DW::Request->get;
+    my $get = $r->get_args;
+
+    my $trust_root = $get->{'openid.trust_root'} // '';
+    my $return_to  = $get->{'openid.return_to'} // '';
+
+    ## Non-OpenID-compliant section: rewrite LiveJournal's trust_root to
+    ## https so that it will match their return_to URL and pass validation.
+    $get->{'openid.trust_root'} = 'https://www.livejournal.com/'
+        if ( $trust_root eq 'http://www.livejournal.com/' &&
```

Comment:
I believe the style guide requests not to use parenthesis when they aren't necessary?